### PR TITLE
Extract reusable id generator function, update `uuid` to `v8.x.x`.

### PR DESCRIPTION
**Pull request by dennisoelkers**
## Motivation and Context

This PR is updating the `uuid` used to generate ids in multiple places to the newest version. For this, we need to change the way the package is imported. In order to reduce future effort to update (or replace) the uuid generation module, this PR is creating a central, reusable id generator function (`generateId`), which is encapsulating the id generation algorithm and the used package.

/jenkins-pr-deps Graylog2/graylog-plugin-enterprise#3476

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/components/event-definitions/event-definition-types/FilterPreviewContainer.jsx [hunk 1]
```diff
@@ -17,21 +17,22 @@
...
 import Search from 'views/logic/search/Search';
 import connect from 'stores/connect';
 import PermissionsMixin from 'util/PermissionsMixin';
 import { CurrentUserStore } from 'stores/users/CurrentUserStore';
-import { FilterPreviewStore, FilterPreviewActions } from 'stores/event-definitions/FilterPreviewStore';
+import { FilterPreviewActions, FilterPreviewStore } from 'stores/event-definitions/FilterPreviewStore';
 
 import FilterPreview from './FilterPreview';
 
+import generateId from '../../../logic/generateId';
```
Comment (Gaya) [inline graylog2-web-interface/src/components/event-definitions/event-definition-types/FilterPreviewContainer.jsx:30]: Relative import?

File: graylog2-web-interface/src/logic/content-packs/ContentPack.js [hunk 2]
```diff
@@ -16,10 +16,11 @@
  */
 import { Map, Set } from 'immutable';
 import { concat, remove } from 'lodash';
-import uuid from 'uuid/v4';
 
 import Entity from './Entity';
 
+import generateId from '../generateId';
```
Comment (Gaya) [inline graylog2-web-interface/src/logic/content-packs/ContentPack.js:22]: Relative import?

**Review by Gaya — APPROVED** [review]
LGTM, though there are some relative imports of `generateId.ts` which I think are meant to be absolute.

File: graylog2-web-interface/src/components/event-definitions/event-definition-types/FilterForm.jsx [hunk 3]
```diff
@@ -35,6 +34,7 @@ import { LookupTablesActions, LookupTablesStore } from 'stores/lookup-tables/Loo
 
 import EditQueryParameterModal from '../event-definition-form/EditQueryParameterModal';
 import commonStyles from '../common/commonStyles.css';
+import generateId from '../../../logic/generateId';
```
[the remark this answers is not in the document]
Comment (dennisoelkers) [reply graylog2-web-interface/src/components/event-definitions/event-definition-types/FilterForm.jsx]: Thanks for catching this! :+1:

File: graylog2-web-interface/src/components/event-definitions/event-definition-types/FilterPreviewContainer.jsx (hunk 1 above)
Comment (dennisoelkers) [reply graylog2-web-interface/src/components/event-definitions/event-definition-types/FilterPreviewContainer.jsx]: Thanks for catching this! :+1:

File: graylog2-web-interface/src/logic/content-packs/ContentPack.js (hunk 2 above)
Comment (dennisoelkers) [reply graylog2-web-interface/src/logic/content-packs/ContentPack.js]: Thanks for catching this! :+1:

**Review by Gaya — APPROVED** [review]
Thanks for fixing the imports. Works great.

**Pull request merged**